### PR TITLE
organizeImports: Avoid using full FindAllReferences

### PR DESCRIPTION
**Pull request by andy-ms**
For imports, all we really need is `getPossibleSymbolReferencePositions` followed by checking for *exact* symbol match. FindAllReferences would look for related symbols, but we don't actually want those in this case.

### CODE REVIEW

File: src/services/findAllReferences.ts [hunk 1]
```diff
@@ -709,6 +709,16 @@ namespace ts.FindAllReferences.Core {
         return exposedByParent ? scope.getSourceFile() : scope;
     }
 
+    /** Used as a quick check for whether a symbol is used at all in a file (besides its definition). */
+    export function isSymbolReferencedInFile(definition: Identifier, checker: TypeChecker, sourceFile: SourceFile) {
```
Comment (amcasey) [inline src/services/findAllReferences.ts]: It probably wouldn't hurt to test this method.

File: src/services/organizeImports.ts [hunk 2]
```diff
@@ -104,23 +104,8 @@ namespace ts.OrganizeImports {
...
-            if (jsxContext && symbol.name === jsxNamespace) {
-                return true;
-            }
-
-            const entries = FindAllReferences.getReferenceEntriesForNode(identifier.pos, identifier, program, [sourceFile], {
-                isCancellationRequested: () => false,
-                throwIfCancellationRequested: () => { /*noop*/ },
-            }).filter(e => e.type === "node" && e.node.getSourceFile() === sourceFile);
-            return entries.length > 1;
+            return jsxContext && identifier.text === jsxNamespace || FindAllReferences.Core.isSymbolReferencedInFile(identifier, typeChecker, sourceFile);
```
Comment (amcasey) [inline src/services/organizeImports.ts]: Personally, I find it helpful to use parentheses when combining binary operators with different precedence.

**Comment by amcasey** [comment]
FYI @mhegazy

File: src/services/findAllReferences.ts (hunk 1 above)
Comment (andy-ms) [reply src/services/findAllReferences.ts]: #22134

**Pull request merged**